daemon/mgr/image_prune.go

```diff
@@ -0,0 +1,123 @@
+package mgr
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alibaba/pouch/apis/filters"
+	"github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/pkg/errtypes"
+	"github.com/alibaba/pouch/pkg/utils"
+
+	"github.com/opencontainers/go-digest"
+	pkgerrors "github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// PruneImage prune unused images
+func (mgr *ImageManager) PruneImage(ctx context.Context, imagesInfo []types.ImageInfo, filter filters.Args) (types.ImagePruneResp, error) {
+	var result types.ImagePruneResp
+
+	if len(imagesInfo) == 0 {
+		return result, nil
+	}
+
+	// 构造返回的list列表，根据filter过滤
+	beforeImages := filter.Get("before")
+	sinceImages := filter.Get("since")
+	referenceFilter := filter.Get("reference")
+
+	// refuse undefined behavior
+	if len(beforeImages) > 1 {
+		return result, pkgerrors.Wrapf(errtypes.ErrInvalidParam, "can't use before filter more than one")
+	}
+	// refuse undefined behavior
```

Comment:
could we reuse the logic in the image filter part?

Reply:
ok